jupyterlab_bigquery/jupyterlab_bigquery/handlers.py

```diff
@@ -0,0 +1,166 @@
+# Lint as: python3
+"""Request handler classes for the extensions."""
+
+import base64
+import json
+import re
+import tornado.gen as gen
+import os
+import random
+
+import json
+import datetime
+
+from collections import namedtuple
+from notebook.base.handlers import APIHandler, app_log
+from google.cloud import bigquery
+
+# import google.auth
+# from google.auth.exceptions import GoogleAuthError
+# from google.auth.transport.requests import Request
+
+from jupyterlab_bigquery.version import VERSION
+
+SCOPE = ("https://www.googleapis.com/auth/cloud-platform",)
+client = bigquery.Client()
```

Comment:
I would recommend structuring this file's bigquery client to be mockable for testing. For example look in https://github.com/GoogleCloudPlatform/jupyter-extensions/blob/master/jupyterlab_automl/jupyterlab_automl/handlers.py

Remove the global client. Then create a function to generate the client like:

```
def create_bigquery_client():
    return bigquery.Client()
```

Change these functions to receive the client, so that we can test these in a later PR:
```
list_projects(client, ...)
list_datasets(client, ...)
list_tables(client, ...)
list_models(client, ...)
get_dataset_details(client, ...)
get_table_details(client, ...)
```

Create the client inside of the ApiHandler functions and pass them into the function.
```
class MyHandler(APIHandler):
  bigquery_client = None

  @gen.coroutine
  def get(self, *args, **kwargs):
     try:
       self.bigquery_client = create_bigquery_client()
       list_projects(self.bigquery_client)
       ...
```